jenkins/bootstrap.py

```diff
@@ -735,19 +737,44 @@ def gubernator_uri(paths):
     return job
 
 
+@contextlib.contextmanager
+def choose_ssh_key(ssh):
+    """Creates a script for GIT_SSH that uses -i ssh if set."""
+    if not ssh:  # Nothing to do
+        yield
+        return
+
+    with tempfile.NamedTemporaryFile(prefix='ssh', delete=False) as fp:
+        fp.write('#!/bin/sh\nssh -i \'%s\' -F /dev/null "${@}"\n' % ssh)
+    try:
+        os.chmod(fp.name, 0500)
+        had = 'GIT_SSH' in os.environ
```

Comment:
yes it is... maybe note what it is used for, since I would not know without google it.